src/com/google/enterprise/adaptor/documentum/DocumentumAdaptor.java

```diff
@@ -967,23 +990,29 @@ protected boolean getGroups(IDfSession session, Principals principals)
         ImmutableSet.Builder<Principal> members = null;
         String groupName = null;
         String objectId = groupsCheckpoint.getObjectId();
-        while (result.next()) {
-          if (!Objects.equals(objectId, result.getString("r_object_id"))) {
-            // We have transitioned to a new group.
-            addGroup(groupName, groups, members, principals);
-            groupsCheckpoint = new Checkpoint(objectId);
-            members = ImmutableSet.builder();
-            groupName = result.getString("group_name");
-            objectId = result.getString("r_object_id");
-            logger.log(Level.FINE, "Found Group: {0}", groupName);
+        try {
+          caughtException = false;
+          while (result.next()) {
+            if (!Objects.equals(objectId, result.getString("r_object_id"))) {
+              // We have transitioned to a new group.
+              addGroup(groupName, groups, members, principals);
+              groupsCheckpoint = new Checkpoint(objectId);
+              members = ImmutableSet.builder();
+              groupName = result.getString("group_name");
+              objectId = result.getString("r_object_id");
+              logger.log(Level.FINE, "Found Group: {0}", groupName);
+            }
+            addMemberPrincipal(members, principals,
+                result.getString("users_names"), false);
+            addMemberPrincipal(members, principals,
+                result.getString("groups_names"), true);
           }
-          addMemberPrincipal(members, principals,
-              result.getString("users_names"), false);
-          addMemberPrincipal(members, principals,
-              result.getString("groups_names"), true);
+          addGroup(groupName, groups, members, principals);
+          groupsCheckpoint = new Checkpoint(stopObjectId);
+        } catch (DfException e) {
+          caughtException = true;
```

Comment:
It seems like we could eliminate this as a field and avoid duplication if we push this up to fillCollection.

Reply:
Changes as discussed.